libpimeval/src/pimUtils.h

```diff
@@ -42,6 +43,20 @@ namespace pimUtils
   std::string getDirectoryPath(const std::string& filePath);
   bool getEnvVar(const std::string &varName, std::string &varValue);
 
+  template<typename T> 
+  T typeCast(const void* inputOperand) {
+    T operandValue = 0;  // Ensure operandValue is initialized
+    T temp;              // Temporary variable of type T
+
+    // Copy the specified number of bytes into temp
+    memcpy(&temp, inputOperand, sizeof(T));
+
+    // Assign the value to operandValue (handles sign extension if T is signed)
+    operandValue = static_cast<T>(temp);
```

Comment:
do we need the `temp` variable at all?
can we not directly copy to the `operandValue`?

Reply:
I am not sure if memcpy will be able to handle the sign extension. If it can't then we can keep it as it is.

Reply:
We need temp for final static_cast which is responsible for sign extension.